### ISSUE TITLE
Define top-level app-wide navigation

### ISSUE THREAD

**Issue by hursey013** [cloud-gov/cg-ui#341]
### Background:
Eventually, the dashboard will be the home to other cloud.gov offerings, and all will be accessed using a cloud.gov account. Providing a consistent and recognizable top-level application nav will help users feel oriented across the app experiences and make essential actions like logging out (or accessing a user profile), returning to the dashboard's home view, and moving between organizations or platform/pages offerings simple and easy. Future offerings are planned beyond the current Dashboard and Pages.

Example design:
![Image](https://github.com/cloud-gov/cg-ui/assets/1178494/007c361b-084e-495e-b751-21d26073dd33)

### Proposed steps
- [x] Review research and understand different states/constraints based on API
- [x] Create mockups representing each state of the interaction 
- [x] Iterate on mockups based on feedback from team
- [x] Collaborate with engineers for technical feasibility of proposed designs
- [ ] Attach final mockups displaying each state with annotations 

_Ensure this issue has a clear Definition of Done in the Acceptance Criteria list below:_ 
### Acceptance Criteria
The proposed design accommodates the following:
- [ ] The design provides context that this is a cloud.gov property through use of brand (logomark and/or colors, etc)
- [ ] There is space to accommodate future navigation items
- [ ] A user can swap between different cloud.gov applications (platform, pages)
- [ ] There's a place for the current org and way to swap between (See related task #340 )
- [ ] Users have the ability to log out using the navigation bar (login happens elsewhere)
- [ ] Handoff to engineering is complete (new issue)

**Comment by beepdotgov** [comment]
Had a fantastic conversation with @sknep to look at five _very early_ navigation concepts, and get a quick temperature check. All five treatments are available in Figma, and attached below in the collapsible section at the end of this comment.

But! After some (excellent) discussion, we agreed that the second option felt like the strongest candidate:

<details>
<summary><strong>Advancing to the next round: Navigation treatment 02</strong></summary>
<img width="1266" alt="Top nav 02" src="https://github.com/user-attachments/assets/ef23cffd-10b2-4144-adad-f17c6dddb773">
<img width="1266" alt="Top nav 02 — orgs open" src="https://github.com/user-attachments/assets/4f63cd28-96b9-4234-881b-267ac5e06b6a">
</details>

Here’s a quick rundown of my notes from our review. (@sknep, if I’ve forgotten or misstated anything, please feel free to correct me!)

**Why this was the strongest candidate.** There are a number of things that felt most successful about this design:

- The “floating” app-wide navigation bars (e.g., in 4 and 5) felt too disconnected from the rest of the design. The “fixed” bar — full-width, and anchored to the [USWDS banner](https://designsystem.digital.gov/components/banner/) — felt most “grounded”, “sturdy”, and “safe.”
- The page’s main content well felt very extensible, and could be potentially applied to other pages/product types.
- The placement of the “choose an organization” dropdown felt the most intentional/correct, especially in the context of all the other tiers of navigation.

**What we’ll change.** We identified a number of items we’d like to revisit in the next round of design:

- Most important: there’s not enough contrast between the Cloud.gov mark (the starred hexagon) and the dark blue background. That distinction needs to be a lot stronger, to make sure the mark’s visible.
- Related to that, some of the palette is working off of old Federalist colors. @sknep reoriented me on a more up-to-date palette; I’ll work on incorporating more of those colors into the next round.
- We’ll remove the rounded edges on various elements, and see how that feels.
- We should ideally have text labels next to the various “product” icons. (i.e., `[ dashboard icon ] Dashboard` `[ pages icon ] Pages`) I’ll work on incorporating those into the next round.

**Open questions.** There are some _non-blocking_ things we might want to discuss:

- What do we call “Dashboard”? We agreed “Cloud” might not be the best option as a product name, if only because it’ll be sitting next to a _Cloud.gov_ logo. But we agreed that the terminology’s okay for now, and we can revisit in the future. (Perhaps informed by user research.)

I’ll get to work on these edits to treatment #2, and share them for review once they’re ready. (Likely tomorrow.)

More soon!

--- 

<details>
<summary><strong>All five initial nav treatments</strong></summary>
<h2>Navigation treatment 01</h2>
<img width="1266" alt="Top nav 01 — default" src="https://github.com/user-attachments/assets/acc39dac-f075-429f-ab95-d2d20bf80815">
<img width="1266" alt="Top nav 01 — org picker open" src="https://github.com/user-attachments/assets/93c35aa8-7465-4932-99fd-dc350e83dace">

<h2>Navigation treatment 02</h2>
<img width="1266" alt="Top nav 02" src="https://github.com/user-attachments/assets/ef23cffd-10b2-4144-adad-f17c6dddb773">

<h2>Navigation treatment 03</h2>
<img width="1266" alt="Top nav 03 — default" src="https://github.com/user-attachments/assets/24064c3f-ff9b-48a6-8693-98080cd5a4e7">
<img width="1266" alt="Top nav 03 — org picker open" src="https://github.com/user-attachments/assets/41cc5288-5bac-4a3d-972c-03d6d53904e2">
<img width="1266" alt="Top nav 03 — products open" src="https://github.com/user-attachments/assets/b0cad21f-6fb9-42c9-8928-bcf8d3f9b989">

<h2>Navigation treatment 04</h2>
<img width="1266" alt="Top nav 04" src="https://github.com/user-attachments/assets/d0d50484-6ac7-49d8-8870-9bcfabf733fe">

<h2>Navigation treatment 05</h2>
<img width="1266" alt="Top nav 05" src="https://github.com/user-attachments/assets/2d63b34a-863d-4b23-b93d-352118914e58">
</details>

**Comment by beepdotgov** [comment]
Had a fantastic check-in with @sknep, and the revised nav feels like our winner! ([Available in Figma](https://www.figma.com/design/DNbFdd7YrA2MpgeXNgmNLz/Cloud.gov-%E2%80%94-Dashboard-UI-redesign?node-id=204-690&t=tRDgFLXnjrgFtOs6-0), but also attached below.)

**Next steps:**

- I’ll work on additional states for the “Current organization” dropdown. (If a user only belongs to a single org, what do they see? If they only belong to (e.g.) two, what does the dropdown look like?)
- There’s some Figma cleanup that needs to happen before handoff, but I don’t think that blocks a technical feasibility review with our engineers. I’ll work on setting that up.

## Mockups
### Revised top nav
<img width="1266" alt="Top nav revised" src="https://github.com/user-attachments/assets/cbe8a33f-efc8-450a-810d-0bbedf324827">

### Revised top nav (small screen)
<img width="398" alt="Top nav revised — small screen" src="https://github.com/user-attachments/assets/9ddc15ff-0aba-4cd1-8a71-55112ab903a1">

### Revised top nav (small screen, nav open)
<img width="398" alt="Top nav revised — small screen, nav open" src="https://github.com/user-attachments/assets/022b1855-02d8-4aa2-8ff1-76b98aeecdd9">

**Comment by beepdotgov** [comment]
Had an excellent technical review with @jduss4 yesterday. I’ve included a couple additional states for the org picker, which should address two additional use cases for #340. Namely:

- What if a user only belongs to one organization?
- What if the user belongs to no organizations? (Very rare/unlikely)

@hursey013: I’d love to do a quick review once you’re back/available, just to make sure these meet all our product requirements. Once we’ve got your thumbs-up, we can move these into implementation.

---

## Single organization
<img width="398" alt="Top nav revised — small screen" src="https://github.com/user-attachments/assets/8f2e67ba-9ca6-420c-827d-6597d65799c4">
<img width="1266" alt="Top nav revised" src="https://github.com/user-attachments/assets/0213ce3b-ba77-4d4f-a1fb-c579b33e8849">

## No organizations
<img width="1266" alt="Top nav revised — no orgs" src="https://github.com/user-attachments/assets/d1e25833-18f3-4afc-9d11-1c81bd69ddcb">

**Comment by beepdotgov** [comment]
(Just formally noting this as blocked for now, pending review later this week.)

**Comment by hursey013** [comment]
@beepdotgov These look GREAT 😍! I agree that #2 will likely provide us with the most flexibility. Visually, I also liked #4, but do see how that option may provide less flexibility as we grow. I think my only feedback with the initial design is something you all already identified - relying solely on icons for navigation is a major pain point with the current solution, Stratos, so making sure we clearly label navigation icons (as you have in your subsequent iterations) is imperative. 

Also wanted to praise the sub-navigation (product sections), I can see how the USWDS nav component can be easily adapted to accomplish the look and feel here, including rolling up into a mobile menu on smaller screens.

I do think the conversation about what to call things is important. `Dashboard` should work for now, but I imagine over time, as we further integrate the platform + pages + build tasks experience, we may need to reconsider what to call things because I imagine eventually there will be some overlap between products. Let's put a pin in that for now though.

Again, amazing work @beepdotgov! I will work with @jduss4 to convert these into implementation issues, but consider this approved, and feel free to unblock/close this issue.

**Comment by beepdotgov** [comment]
_Fantastic_ news all around, @hursey013. Thanks so much for taking a look, and for all the feedback. So glad to hear we’re on the right track.

I’ll move these artboards into the “Build” section of the Figma board. If there’s anything I can provide you and @jduss4 to help with implementation, please let me know.

Excited to close this out. Onward! 🚀
